src/lib.rs

```diff
@@ -0,0 +1,62 @@
+use std::io::Read;
+
+/// List of errors that the program can return.
+#[derive(Debug)]
+pub enum DepotError {
+    UnknownOperatingSystem,
+}
+
+/// Result type wich wither take a type T or a DepotError.
+pub type DepotResult<T> = std::result::Result<T, DepotError>;
+pub fn unwrap_depot_error<T>(result: DepotResult<T>) -> T {
+    match result {
+        Ok(value) => value,
+        Err(error) => {
+            eprintln!(
+                "{}",
+                match error {
+                    DepotError::UnknownOperatingSystem =>
+                        "Unable to determine your current operating system.",
+                }
+            );
+            std::process::exit(1);
+        }
+    }
+}
+
+/// List of all supported operating systems.
+#[derive(Debug)]
+pub enum OperatingSystem {
+    Arch,
+    Alpine,
+    Debian,
+    Ubuntu,
+    Fedora,
+}
+impl OperatingSystem {
+    /// Get the currently running operating system.
+    pub fn current() -> DepotResult<OperatingSystem> {
+        match std::env::consts::OS {
+            "linux" => {
+                let mut contents = String::new();
+                std::fs::File::open("/etc/os-release")
+                    .unwrap()
+                    .read_to_string(&mut contents)
+                    .unwrap();
+                match contents
+                    .split('\n')
+                    .find(|line| line.starts_with("NAME="))
+                    .unwrap()
+                {
+                    "NAME=\"Arch Linux\"" => Ok(OperatingSystem::Arch),
+                    "NAME=\"Alpine Linux\"" => Ok(OperatingSystem::Alpine),
+                    "NAME=\"Debian GNU/Linux\"" => Ok(OperatingSystem::Debian),
+                    "NAME=\"Ubuntu\"" => Ok(OperatingSystem::Ubuntu),
+                    "NAME=\"Fedora Linux\"" => Ok(OperatingSystem::Fedora),
+                    _ => Err(DepotError::UnknownOperatingSystem),
+                }
+            }
```

Comment:
I think it will be better if you use ID instead of name. Name is more for end users than for devs

Reply:
doesn't id change with version?

Reply:
can be done, I didn't saw it when I first looked for a way to get the os